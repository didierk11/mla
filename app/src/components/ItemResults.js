import React from "react";
import { formatPrice } from "../utils/utils";

const ItemResults = (props) => {
  const {
    id,
    picture: src,
    title,
    free_shipping,
    state,
    amount,
    currency,
    decimals,
  } = props;

  return (
    <div className="row justify-content-md-center item-list-row">
      <div className="col-lg-2">
        <div className="item-list-img-container">
          <div className="item-list-img">
            <img alt="Imagen del producto" src={src}></img>
          </div>
        </div>
      </div>
      <div className="col-lg-10">
        <div className="item-list">
          <div className="row">
            <div className="col-md-8">
              <div className="item-list-data">
                <div className="item-list-price">
                  $ {formatPrice(amount, currency, "decimal")}
                  {decimals > 0 && (
                    <sup className="item-list-price-decinal">.{decimals}</sup>
                  )}{" "}
                  {free_shipping && (
                    <span>
                      <div className="item-list-free-shipping-img"></div>
                    </span>
                  )}
                </div>
                <div>
                  <a href={`/items/${id}`}>{title}</a>
                </div>
              </div>
            </div>
            <div className="col-md-4">
              <div className="item-list-state">{state}</div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ItemResults;
