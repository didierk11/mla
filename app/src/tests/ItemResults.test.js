import React from "react";
import { render } from "react-dom";
import ItemResults from "../components/ItemResults";
import renderer from "react-test-renderer";
import data from "./dataSets/items.json";

// data
const {id, src, title, free_shipping, state} = data;
const {amount, currency, decimals} = data.price;

// test
describe("Snapshot test:", () => {
  it("ItemResults render correctly", () => {
    const div = renderer
      .create(
        <ItemResults
          id={id}
          src={src}
          free_shipping={free_shipping}
          state={state}
          title={title}
          amount={amount}
          currency={currency}
          decimals={decimals}
        />
      )
      .toJSON();
    expect(div).toMatchSnapshot();
  });
});

describe("Smoke test:", () => {
  it("ItemResults render without crashing", () => {
    const div = document.createElement("div");
    render(
      <ItemResults
        id={id}
        src={src}
        free_shipping={free_shipping}
        state={state}
        title={title}
        amount={amount}
        currency={currency}
        decimals={decimals}
      />,
      div
    );
  });
});
