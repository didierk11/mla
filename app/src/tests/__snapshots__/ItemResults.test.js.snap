// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test: ItemResults render correctly 1`] = `
<div
  className="row justify-content-md-center item-list-row"
>
  <div
    className="col-lg-2"
  >
    <div
      className="item-list-img-container"
    >
      <div
        className="item-list-img"
      >
        <img
          alt="Imagen del producto"
        />
      </div>
    </div>
  </div>
  <div
    className="col-lg-10"
  >
    <div
      className="item-list"
    >
      <div
        className="row"
      >
        <div
          className="col-md-8"
        >
          <div
            className="item-list-data"
          >
            <div
              className="item-list-price"
            >
              $ 
              949
              <sup
                className="item-list-price-decinal"
              >
                .
                9
              </sup>
               
              <span>
                <div
                  className="item-list-free-shipping-img"
                />
              </span>
            </div>
            <div>
              <a
                href="/items/MLA869265149"
              >
                Pelota De Futbol Nº 5 + Calidad 350 Gramos Cámara De Látex
              </a>
            </div>
          </div>
        </div>
        <div
          className="col-md-4"
        >
          <div
            className="item-list-state"
          >
            Buenos Aires
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
